vllm/engine/metrics.py

```diff
@@ -133,7 +133,37 @@ def __init__(self, labelnames: List[str], max_model_len: int):
             name="vllm:e2e_request_latency_seconds",
             documentation="Histogram of end to end request latency in seconds.",
             labelnames=labelnames,
-            buckets=[1.0, 2.5, 5.0, 10.0, 15.0, 20.0, 30.0, 40.0, 50.0, 60.0])
+            buckets=[
+                0.3, 0.5, 0.8, 1.0, 1.5, 2.0, 2.5, 5.0, 10.0, 15.0, 20.0, 30.0,
+                40.0, 50.0, 60.0
+            ])
+        self.histogram_time_in_queue_request = self._histogram_cls(
+            name="vllm:time_in_queue_requests",
+            documentation=
+            "Histogram of time the request spent in the queue in seconds.",
+            labelnames=labelnames,
+            buckets=[
+                0.3, 0.5, 0.8, 1.0, 1.5, 2.0, 2.5, 5.0, 10.0, 15.0, 20.0, 30.0,
+                40.0, 50.0, 60.0
+            ])
+        self.histogram_model_forward_time_request = self._histogram_cls(
+            name="vllm:model_forward_time_seconds",
+            documentation=
+            "Histogram of time spent in the model forward pass in seconds.",
+            labelnames=labelnames,
+            buckets=[
+                0.3, 0.5, 0.8, 1.0, 1.5, 2.0, 2.5, 5.0, 10.0, 15.0, 20.0, 30.0,
+                40.0, 50.0, 60.0
+            ])
+        self.histogram_model_execute_time_request = self._histogram_cls(
+            name="vllm:model_execute_time_seconds",
+            documentation=
+            "Histogram of time spent in the model execute function in seconds.",
+            labelnames=labelnames,
+            buckets=[
+                0.3, 0.5, 0.8, 1.0, 1.5, 2.0, 2.5, 5.0, 10.0, 15.0, 20.0, 30.0,
+                40.0, 50.0, 60.0
+            ])
```

Comment:
The model forward and execution typically runs in the order of ms not s. This histogram bucket is not every informative. I suggest (1) keep it unit of ms (2) bucket from 2ms to 200ms

Reply:
> The model forward and execution typically runs in the order of ms not s. This histogram bucket is not every informative. I suggest (1) keep it unit of ms (2) bucket from 2ms to 200ms

Thank you for your reply. In my actual testing, I found that the latency of forward and execute sometimes reaches 800ms, so I increased the value range from 1ms to 3000ms to accommodate different application scenarios.